src/contracts/core/RewardsCoordinator.sol

```diff
@@ -430,30 +564,104 @@ contract RewardsCoordinator is
+        address earner = claim.earnerLeaf.earner;
+        address claimer = claimerFor[earner];
+        if (claimer == address(0)) {
+            claimer = earner;
+        }
+        require(msg.sender == claimer, "RewardsCoordinator.processClaim: caller is not valid claimer");
+        for (uint256 i = 0; i < claim.tokenIndices.length; i++) {
+            TokenTreeMerkleLeaf calldata tokenLeaf = claim.tokenLeaves[i];
+
+            uint256 currCumulativeClaimed = cumulativeClaimed[earner][tokenLeaf.token];
+            require(
+                tokenLeaf.cumulativeEarnings > currCumulativeClaimed,
+                "RewardsCoordinator.processClaim: cumulativeEarnings must be gt than cumulativeClaimed"
+            );
+
+            // Calculate amount to claim and update cumulativeClaimed
+            uint256 claimAmount = tokenLeaf.cumulativeEarnings - currCumulativeClaimed;
+            cumulativeClaimed[earner][tokenLeaf.token] = tokenLeaf.cumulativeEarnings;
+
+            tokenLeaf.token.safeTransfer(recipient, claimAmount);
+            emit RewardsClaimed(root.root, earner, claimer, recipient, tokenLeaf.token, claimAmount);
+        }
+    }
+
     function _setActivationDelay(uint32 _activationDelay) internal {
         emit ActivationDelaySet(activationDelay, _activationDelay);
         activationDelay = _activationDelay;
     }
 
-    function _setGlobalOperatorCommission(uint16 _globalCommissionBips) internal {
-        emit GlobalCommissionBipsSet(globalOperatorCommissionBips, _globalCommissionBips);
-        globalOperatorCommissionBips = _globalCommissionBips;
+    function _setDefaultOperatorSplit(uint16 split) internal {
+        emit DefaultOperatorSplitBipsSet(defaultOperatorSplitBips, split);
+        defaultOperatorSplitBips = split;
     }
 
     function _setRewardsUpdater(address _rewardsUpdater) internal {
         emit RewardsUpdaterSet(rewardsUpdater, _rewardsUpdater);
         rewardsUpdater = _rewardsUpdater;
     }
 
+    /**
+     * @notice Internal helper to set the operator split.
+     * @param operatorSplit The split struct for an Operator
+     * @param split The split in basis points.
+     * @param activatedAt The timestamp when the split is activated.
+     */
+    function _setOperatorSplit(OperatorSplit storage operatorSplit, uint16 split, uint32 activatedAt) internal {
+        // If, the earlier 'new' split is activated, we update the 'old' split with the earlier 'new' split.
+        // Else, the earlier 'old' split remains the same. This is essentially resetting the activation delay window
+        // since the earlier split setting didn't complete.
+        if (block.timestamp >= operatorSplit.activatedAt) {
+            if (operatorSplit.activatedAt == 0) {
+                // If the operator split has not been initialized yet, set the old split to the default split.
+                operatorSplit.oldSplitBips = defaultOperatorSplitBips;
```

Comment:
Added the following condition for the edge case of getting correct operator split before the first time the split is activated. Discovered it while writing test cases.